src/main/kotlin/org/jetbrains/bio/statistics/mixture/Internals.kt

```diff
@@ -14,7 +14,7 @@ import org.jetbrains.bio.viktor._I
 object MixtureInternals {
     @JvmStatic
     fun predict(logGammas: F64Array): IntArray {
-        return IntArray(logGammas.shape[0]) { logGammas.V[_I, it].argMax() }
+        return IntArray(logGammas.shape[1]) { logGammas.V[_I, it].argMax() }
```

Comment:
It was, I even commented on it, and you asked me to provide a test. Since this is a really hard-to-miss bug (shape is usually something like 3 x 15_000_000), I'd say that the integration and sample-and-fit tests that we have cover this issue pretty well.
(That reminds me, though, that I've recently killed `argMax` in `viktor`. I have to either bring it back or work around this.)